src/main.cpp

```diff
@@ -300,6 +296,97 @@ CNodeState *State(NodeId pnode) {
     return &it->second;
 }
 
+class CMisbehaviorTracker {
+    typedef std::lock_guard<std::mutex> lock_guard;
+    struct Info {
+        const std::string name;
+        int badness;
+        bool ban;
+        Info(std::string&& nameIn, int badnessIn, bool banIn) : name(std::move(nameIn)), badness(badnessIn), ban(banIn) {}
```

Comment:
Thanks for clearing that up, I'm still learning all the details related to rvalue reference semantics. I'm starting to understand the necessity of all the logic related to forwarding arguments...